analyzer/src/main/kotlin/managers/Bundler.kt

```diff
@@ -35,4 +71,192 @@ class Bundler : PackageManager() {
+        var tempVendorDir: File? = null
+
+        try {
+            if (vendorDir.isDirectory) {
+                val tempDir = createTempDir("analyzer", ".tmp", workingDir)
+                tempVendorDir = File(tempDir, "vendor")
+                log.warn { "'$vendorDir' already exists, temporarily moving it to '$tempVendorDir'." }
+                Files.move(vendorDir.toPath(), tempVendorDir.toPath(), StandardCopyOption.ATOMIC_MOVE)
+            }
+            val scopes = mutableSetOf<Scope>()
+            val packages = mutableSetOf<Package>()
+            val errors = mutableListOf<String>()
+            val vcsDir = VersionControlSystem.forDirectory(workingDir)
+
+            installDependencies(workingDir)
+
+            val (projectName, version, projectHomepageUrl, declaredLicenses) = parseProject(workingDir)
+            val groupedDeps = getDependencyGroups(workingDir)
+
+            for ((groupName, dependencyList) in groupedDeps) {
+                parseScope(workingDir, groupName, dependencyList, scopes, packages, errors)
+            }
+
+            val project = Project(
+                    id = Identifier(
+                            provider = javaClass.simpleName,
+                            namespace = "",
+                            name = projectName,
+                            version = version
+                    ),
+                    declaredLicenses = declaredLicenses.toSortedSet(),
+                    aliases = emptyList(),
+                    vcs = vcsDir?.getInfo(workingDir) ?: VcsInfo.EMPTY,
+                    homepageUrl = projectHomepageUrl,
+                    scopes = scopes.toSortedSet()
+            )
+            return AnalyzerResult(
+                    true,
+                    project,
+                    packages.map { it.toCuratedPackage() }.toSortedSet(),
+                    errors
+            )
+        } finally {
+            // Delete vendor folder to not pollute the scan.
+            if (!vendorDir.deleteRecursively()) {
+                throw IOException("Unable to delete the '$vendorDir' directory.")
+            }
+
+            // Restore any previously existing "vendor" directory.
+            if (tempVendorDir != null) {
+                log.info { "Restoring original '$vendorDir' directory from '$tempVendorDir'." }
+                Files.move(tempVendorDir.toPath(), vendorDir.toPath(), StandardCopyOption.ATOMIC_MOVE)
+                if (!tempVendorDir.parentFile.delete()) {
+                    throw IOException("Unable to delete the '${tempVendorDir.parent}' directory.")
+                }
+            }
+        }
+    }
+
+    private fun parseScope(workingDir: File, groupName: String, dependencyList: List<String>, scopes: MutableSet<Scope>,
+                           packages: MutableSet<Package>, errors: MutableList<String>) {
+        log.debug("Parsing scope: $groupName\nscope top level deps list=$dependencyList")
```

Comment:
Please use curly brace syntax for log statements:

    log.debug { "Parsing scope: $groupName\nscope top level deps list=$dependencyList" }

(This avoid the string to get evaluated / expanded if debug logging it not enabled.)